jiant/scripts/postproc/xtreme/xtreme_runconfig_writer.py

```diff
@@ -0,0 +1,114 @@
+import jiant.proj.main.scripts.configurator as configurator
+import jiant.utils.zconf as zconf
+import jiant.utils.python.io as py_io
+from jiant.utils.python.datastructures import get_unique_list_in_order
+
+
+LANGS_DICT = {
+    "xnli": "ar bg de el en es fr hi ru sw th tr ur vi zh".split(),
+    "pawsx": "de en es fr ja ko zh".split(),
+    "udpos": "af ar bg de el en es et eu fa fi fr he hi hu id it ja ko mr"
+    " nl pt ru ta te tr ur vi zh".split(),
+    "panx": "af ar bg bn de el en es et eu fa fi fr he hi hu id it ja jv ka kk ko ml mr ms my"
+    " nl pt ru sw ta te th tl tr ur vi yo zh".split(),
+    "xquad": "ar de el en es hi ru th tr vi zh".split(),
+    "mlqa": "ar de en es hi vi zh".split(),
+    "tydiqa": "ar bn en fi id ko ru sw te".split(),
+    "bucc2018": "de fr ru zh".split(),
+    "tatoeba": "af ar bg bn de el es et eu fa fi fr he hi hu id it ja jv ka kk ko ml mr"
+    " nl pt ru sw ta te th tl tr ur vi zh".split(),
+}
+TRAIN_TASK_DICT = {
+    "xnli": "mnli",
+    "pawsx": "pawsx_en",
+    "udpos": "udpos_en",
+    "panx": "panx_en",
+    "xquad": "squad_v1",
+    "mlqa": "squad_v1",
+    "tydiqa": "tydiqa_en",
+}
+TRAINED_TASKS = ["xnli", "pawsx", "udpos", "panx", "xquad", "mlqa", "tydiqa"]
+UNTRAINED_TASKS = ["bucc2018", "tatoeba"]
+
+
+@zconf.run_config
+class RunConfiguration(zconf.RunConfig):
+    xtreme_task = zconf.attr(type=str, required=True)
+    task_config_base_path = zconf.attr(type=str, required=True)
+    task_cache_base_path = zconf.attr(type=str, required=True)
+    output_path = zconf.attr(type=str, required=True)
+
+    # Optional
+    epochs = zconf.attr(type=int, default=3)
+    train_batch_size = zconf.attr(type=int, default=4)
+    eval_batch_multiplier = zconf.attr(type=int, default=2)
+    gradient_accumulation_steps = zconf.attr(type=int, default=1)
+    eval_subset_num = zconf.attr(type=int, default=500)
+    num_gpus = zconf.attr(type=int, default=1)
+    early_stop_on_xtreme_tasks = zconf.attr(
+        action="store_true",
+        help="False = Do early stopping on train task,"
+        " True = Do early stopping on XTREME tasks in all languages"
+        " (default: False)",
+    )
+    no_verbose = zconf.attr(action="store_true")
+
+
+def generate_configs(args: RunConfiguration):
+    xtreme_task = args.xtreme_task
+    if xtreme_task == "mlqa":
+        xtreme_task_name_list = [f"{xtreme_task}_{lang}_{lang}" for lang in LANGS_DICT[xtreme_task]]
```

Comment:
Is this a quirk of the downloaded data? Would prefer to fix this early on.

Reply:
Yes, most XTREME tasks have just a single language, or are a single language paired with English, but MLQA actually has data for every pair. However, within XTREME, we only care about pairs like `mlqa_zh_zh`.